Extended controller apps tests

asmodehn opened the pull request:
## Checklist

- [X] I have read the [OpenWISP Contributing Guidelines](http://openwisp.io/docs/developer/contributing.html).
- [X] I have manually tested the changes proposed in this pull request.
- [X] I have written new test cases for new code and/or updated existing tests for changes to existing code.
- [ ] I have updated the documentation.

## Reference to Existing Issue

Closes #298 

## Description of Changes

Added sample_config and sample_connection in tests.
Adjust the tests to pass (OK on my machine, still WIP for CI...)

nemesifier reviewed (commented):
@asmodehn there's a conflict with master.

asmodehn commented:
@nemesifier From what I can tell, these test failures come from `sample_config` and `sample_connection` tests that I have added to demonstrate another problem...

However these were not tested before... and the issue seems somewhat similar to https://github.com/openwisp/openwisp-controller/issues/442, by which I mean "controller tests fail when accompanied by firmware_upgrader", albeit for other reasons...

So, I see few possible outcomes for this pull request:
- keep fixing everything in this PR (it would likely take some time, and synchronization with controller changes...)
- remove the tests from sample_config and sample_connection (but I feel we should keep them and actually get them to work eventually...)
- only test `openwisp2.sample_firmware_upgrader` with `SAMPLE_APP=1` (as it was before actually) by specifying it in `./runtests.py`, and create an issue in here/in controller to get controller tests to work with firmware_upgrader, and deal with that as a different issue.

I'd favor the latter, but let me know which way you want to go, or if you see another possible option ?

Sadly, I don't think I will be very resourceful to fix these controller failing tests. The errors are regarding the database usage of the controller, and I don't know enough about this topic...

nemesifier reviewed (commented):
@asmodehn can you update your openwisp-utils to the current latest master and run `openwisp-qa-format` on your branch and update this PR please?

This is due to:

https://github.com/openwisp/openwisp-utils/pull/456/
https://github.com/openwisp/openwisp-utils/commit/c0bde3cec6611518d71ff60c49765f1b0a16e6f8

asmodehn commented:
@nemesifier Somehow openwisp-qa-format didn't do the job, I had to run black manually to replace quotes... I m not sure why that was the case...

Commit:
[fix] QA-checks and format

nemesifier commented:
> @nemesifier Somehow openwisp-qa-format didn't do the job, I had to run black manually to replace quotes... I m not sure why that was the case...

Make sure your openwisp-utils is up to date with the latest master, you may need to uninstall it and reinstall it.

Commit:
[fix] Add floorplan.jpg to fix test_connection_upload

Commit:
[tests] Add sample_config and sample_connection

aim is to detect problems when using extended controller

# Conflicts:
#	tests/openwisp2/settings.py
#	tests/openwisp2/urls.py

Commit:
[tests] Fix firmware_upgrader tests with sample_config & sample_connection

Commit:
[change] QA-format with openwisp-utils@ba491d4

Commit:
[change] Formatting

nemesifier committed:
Merge branch 'master' into extended_controller_apps_tests
